edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/EdgeHubProtocolHead.cs

```diff
@@ -31,7 +33,15 @@ public Task StartAsync()
         public Task CloseAsync(CancellationToken token)
         {
             this.logger.LogInformation($"Closing protocol heads - {this.Name}");
-            return Task.WhenAll(this.underlyingProtocolHeads.Select(protocolHead => protocolHead.CloseAsync(token)));
+            try
+            {
+                return Task.WhenAll(this.underlyingProtocolHeads.Select(protocolHead => protocolHead.CloseAsync(token))).TimeoutAfter(TimeoutInSecs);
```

Comment:
should you use TaskEx.TimeoutAfter here too until we find out why task.TimeoutAfter doesn't work?

Reply:
task.TimeoutAfter doesn't work?! Why?